File: samples/oneclickbusiness/app/src/main/java/com/operatortokenocb/network/PartnerManagementApi.java
package com.operatortokenocb.network;

import io.reactivex.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface PartnerManagementApi {

    @GET("access-token/{credentials}")
    Observable<String> observeAccessToken(@Path("credentials") String credentials);

    @POST("bearer-token-hackathon")
    Observable<String> observeBearerToken(@Body GetBearerBody body);
}

File: SmartCredentials_aOS/identityprovider/src/main/java/de/telekom/identityprovider/rest/PartnerManagementApi.kt
/*
 * Copyright (c) 2019 Telekom Deutschland AG
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package de.telekom.identityprovider.rest

import io.reactivex.Observable
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

/**
 * Created by teodorionut.ganga@endava.com at 23/02/2023
 */
interface PartnerManagementApi {

    @GET("access-token/{credentials}")
    fun observeAccessToken(@Path("credentials") credentials: String): Observable<String>

    @POST("bearer-token-hackathon")
    fun observeBearerToken(@Body body: GetBearerBody): Observable<String>
}